RFCs/005-Framework-Extensibility-Custom-DataSource.md

```diff
@@ -0,0 +1,59 @@
+# RFC 005- Framework Extensibility for Custom DataSource
+
+## Summary
+This details the MSTest V2 framework extensibility for specifying custom data source for data driven tests.
+
+## Motivation
+Often times, custom data source is required for data driven tests. If the test framework provides an extensibility, custom data source functionality can be
+* Easily extended to use test data from custom data sources.
+
+## Detailed Design
+
+### Requirements
+1. Test data from custom data source should be easily associated with multiple test cases.
+
+### Proposed solution
+Here is a solution using which test data can be easily assocaited with test case.
+
+The test framework should define an interface `IDataSource` which can be extended to get data from custom data source. The defintion of this interface is as below :
+```
+    public interface IDataSource
+    {
+        /// <summary>
+        /// Gets the data from custom data source.
+        /// </summary>
+        public IEnumerable<object[]> GetData();
+    }
+```
+
+Here is how the test methods are decorated with concrete implementation of `IDataSource`:
+```
+    public class CustomTestDataSourceAttribute : IDataSource, Attribute
+    {
+        public IEnumerable<object[]> GetData()
+        {
+            return new[]
+                    {   new object[] {1, 2, 3},
+                        new object[] {4, 5, 6}
+                    };
+        }
+    }
+
+    [TestMethod]
```

Comment:
Currently, `DataRow` can be decorated with both `DataTestMethod` and `TestMethod` attributes.

However, we should only support `DynamicData` with `DataTestMethod`. This could be a logical distinction between data-driven tests and normal tests.

From performance perspective also, this approach will be better as we need not to probe for `DynamicData` (or any other attribute inheriting from `IDataSource` while executing normal tests.